### DISCUSSION TITLE
Example of Elasticsearch Pass-Through query

### DISCUSSION THREAD

**Discussion by metalshanked** [trinodb/trino#12341] [Q&A] [answered]
Hi,
Apologies for this basic question
Can anyone provide an example of using a passthrough query in Elasticsearch?
https://trino.io/docs/current/connector/elasticsearch.html#pass-through-queries

I am confused on what to pass as the query DSL
When i try base32 encoding my DSL query, no results show up
`SELECT * FROM es.default."<index>$query:<es-query>"`

Thanks in advance!

**Comment by metalshanked** [comment, marked as answer]
Update: Found the answer, I had an extra colon earlier which was causing empty results

Example: Standard DSL Query
```
GET my_index/_search
{
  "query": {
    "query_string": {
      "fields": ["message"],
      "query": "Hello World"
    }
  }
}
```

Convert above query (just the query part and not GET my_index/_search ) into Base32 using any web or code tool (easily available)

Converted BASE32 string 
```
PMFCAIBCOF2WK4TZEI5CA6YKEAQCAIBCOF2WK4TZL5ZXI4TJNZTSEORAPMFCAIBAEAQCAITGNFSWYZDTEI5CAWZCNVSXG43BM5SSEXJMBIQCAIBAEAQCE4LVMVZHSIR2EAREQZLMNRXSAV3POJWGIIQKEAQCAID5BIQCA7IKPU======
```

Use in Query
```
SELECT * FROM es.default."my_index$query:PMFCAIBCOF2WK4TZEI5CA6YKEAQCAIBCOF2WK4TZL5ZXI4TJNZTSEORAPMFCAIBAEAQCAITGNFSWYZDTEI5CAWZCNVSXG43BM5SSEXJMBIQCAIBAEAQCE4LVMVZHSIR2EAREQZLMNRXSAV3POJWGIIQKEAQCAID5BIQCA7IKPU======"
```

**Comment by findepi** [comment]
I suppose https://github.com/trinodb/trino/pull/12324 will make these things a lot easier.
cc @kasiafi
